storage/metric_directory.go

```diff
@@ -39,24 +41,43 @@ func (md *MetricDirectory) generateParentDirectories() []MetricDirectory {
 	return parents
 }
 
+// UpdateDirectories is a recursive function that will test missing directories
+// and insert them
 func (md *MetricDirectory) UpdateDirectories(storageConnector BgMetadataStorageConnector) error {
 	entry, err := storageConnector.SelectDirectory(md.name)
 	if err != nil {
-		if entry == "" {
-			parentDirectories := md.generateParentDirectories()
-			for _, pd := range parentDirectories {
-				err = storageConnector.InsertDirectory(pd)
-				if err != nil {
-					fmt.Println(err.Error())
-				}
+		if entry != "" {
+			p := NewMetricDirectory(md.parent)
+			p.UpdateDirectories(storageConnector)
+			err = storageConnector.InsertDirectory(md)
+			if err != nil {
+				return fmt.Errorf("cannot insert metric directory: %s", err.Error())
 			}
-		} else {
-			return fmt.Errorf("cannot check metric directory in cassandra: %s", err.Error())
 		}
 	}
-	err = storageConnector.InsertDirectory(*md)
-	if err != nil {
-		fmt.Println(err.Error())
-	}
 	return nil
 }
+
+// getUUID returns a hash of the name
+func (md *MetricDirectory) getUUID() string {
+	namespace, _ := uuid.Parse("00000000-1111-2222-3333-444444444444")
```

Comment:
Why is there a dummy value here ?

Reply:
It is the namespace, it is needed to match biggraphite UUID